### DISCUSSION TITLE
How to include all locales?

### DISCUSSION THREAD

**Discussion by web-programmer-here** [js-fns/date-fns#2719] [Q&A] [answered]
How can I include all locales?
something like this should work but it doesnt

```
import * from 'date-fns/locale'
```

what I want to do is import all locales and pass the language 
moment has this option 

```
moment.locale('es')
```

my application has different languages and I want to pass language codes to date-fns

**Comment by fturmel** [comment, marked as answer]
You need to name the wildcard export if you are to reference it in your code. If this is client-side, keep in mind that this might significantly affect your bundle size.

```js
import * as locales from 'date-fns/locale'

console.log(locales.enUS.code) // en-US
```

**Reply by web-programmer-here** [reply]
is there a cdn version for individual languages?

**Reply by web-programmer-here** [reply]
It says "moment-with-locales.min.js" is only 73 k

<img width="902" alt="image" src="https://user-images.githubusercontent.com/81479051/138564862-bcecc585-ac6d-4d0f-81de-339e20708790.png">

is something like this available for date-fns?

**Reply by fturmel** [reply]
> is there a cdn version for individual languages?

No official CDN as far as I know, and the locales are not published separately. You could try jsDelivr or Skypack and see if they work for you:
https://www.jsdelivr.com/package/npm/date-fns
https://www.skypack.dev/view/date-fns

> It says "moment-with-locales.min.js" is only 73 k. is something like this available for date-fns?

`import * as locales from 'date-fns/locale'` is essentially that; you'd be bundling every locale in the library whether it's used or not. Locales are each approx 1.5k gzipped and there are currently 83. Nothing is stopping you from including all of them, but there is a filesize cost that comes with doing this.

Moment doesn't support tree shaking, so it's all or nothing. That's one of the main reasons the library is being officially deprecated, per https://momentjs.com/docs/#/-project-status/.
